WordPressAuthenticator/Signin/Login2FAViewController.swift

```diff
@@ -283,19 +284,39 @@ class Login2FAViewController: LoginViewController, NUXKeyboardResponder, UITextF
 
 
     @objc func applicationBecameInactive() {
-        pasteboardBeforeBackground = UIPasteboard.general.string
+        pasteboardChangeCountBeforeBackground = UIPasteboard.general.changeCount
     }
 
 
     @objc func applicationBecameActive() {
         let emptyField = verificationCodeField.text?.isEmpty ?? true
         guard emptyField,
-            let pasteString = UIPasteboard.general.string,
-            pasteString != pasteboardBeforeBackground else {
+            pasteboardChangeCountBeforeBackground != UIPasteboard.general.changeCount else {
                 return
         }
 
-        switch isValidCode(code: pasteString) {
+        if #available(iOS 14.0, *) {
+            UIPasteboard.general.detect(patterns: [.number]) { [weak self] result in
+                switch result {
+                case .success(let detections):
+                    if let pasteString = detections.first?.value as? String {
+                        DispatchQueue.main.async {
```

Comment:
We could dispatch the call to the completion block in our category, so we don't need to do it here or in the other view controller?

Reply:
I usually prefer to keep the Main queue dispatch at the call site since other users might not want to dispatch to main right away.

Since this is only used in one spot, though, I'll just make the change in the two success blocks.